src/test/kotlin/com/pubnub/api/subscribe/eventengine/effect/HandshakeReconnectEffectTest.kt

```diff
@@ -0,0 +1,136 @@
+package com.pubnub.api.subscribe.eventengine.effect
+
+import com.pubnub.api.PubNubException
+import com.pubnub.api.endpoints.remoteaction.RemoteAction
+import com.pubnub.api.subscribe.eventengine.event.Event
+import com.pubnub.api.subscribe.eventengine.event.SubscriptionCursor
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import java.time.Duration
+import java.util.concurrent.Executors
+
+class HandshakeReconnectEffectTest {
+    private val channels = listOf("channel1")
+    private val channelGroups = listOf("channelGroup1")
+    private val reason = PubNubException("Unknown error")
+    private val attempts = 1
+    private val handshakeProvider = mockk<HandshakeProvider>()
+    private val eventDeliver = SubscribeManagedEffectFactoryTest.TestEventDeliver()
+    private val handshakeReconnectInvocation =
+        SubscribeEffectInvocation.HandshakeReconnect(channels, channelGroups, attempts, reason)
+    private val executorService = Executors.newSingleThreadScheduledExecutor()
+    private val policy = mockk<RetryPolicy>()
```

Comment:
Instead of setting the mocks everywhere for it why not use the real code and if necessary use other already written policy?

```suggestion
    private val policy = LinearPolicy(duration = Duration.ofMillis(10))
```

Reply:
Yes. I can do this.